From a9c3de71899ffee02dbc6b2b33d865d22a073432 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Thu, 2 Apr 2020 15:37:54 +0200
Subject: [PATCH 28/37] libarchive: don't force the Windows version to target

We may want to build for Win10/UWP via the command-line.
---
 ...ows-versions-if-they-are-set-in-the-.patch | 44 +++++++++++++++++++
 contrib/src/libarchive/rules.mak              |  1 +
 2 files changed, 45 insertions(+)
 create mode 100644 contrib/src/libarchive/0005-don-t-force-windows-versions-if-they-are-set-in-the-.patch

diff --git a/contrib/src/libarchive/0005-don-t-force-windows-versions-if-they-are-set-in-the-.patch b/contrib/src/libarchive/0005-don-t-force-windows-versions-if-they-are-set-in-the-.patch
new file mode 100644
index 0000000000..37af69a79a
--- /dev/null
+++ b/contrib/src/libarchive/0005-don-t-force-windows-versions-if-they-are-set-in-the-.patch
@@ -0,0 +1,44 @@
+From a4fa6bef8bdb141d629d79d8d1f3e67448bd67e1 Mon Sep 17 00:00:00 2001
+From: Steve Lhomme <robux4@ycbcr.xyz>
+Date: Thu, 2 Apr 2020 15:33:40 +0200
+Subject: [PATCH 5/5] don't force windows versions if they are set in the
+ environment
+
+---
+ configure.ac | 20 +++++++++++++++++---
+ 1 file changed, 17 insertions(+), 3 deletions(-)
+
+diff --git a/configure.ac b/configure.ac
+index 7312c4d5..05ca4955 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -244,9 +244,23 @@ AM_CONDITIONAL([STATIC_BSDCPIO], [ test "$static_bsdcpio" = yes ])
+ # Set up defines needed before including any headers
+ case $host in
+   *mingw* | *cygwin* | *msys*  )
+-  AC_DEFINE([_WIN32_WINNT], 0x0502, [Define to '0x0502' for Windows Server 2003 APIs.])
+-  AC_DEFINE([WINVER], 0x0502, [Define to '0x0502' for Windows Server 2003 APIs.])
+-  AC_DEFINE([NTDDI_VERSION], 0x05020000, [Define to '0x05020000' for Windows Server 2003 APIs.])
++  AC_PREPROC_IFELSE([AC_LANG_PROGRAM(
++    [[#ifdef _WIN32_WINNT
++     # error _WIN32_WINNT already defined
++     #endif
++    ]],[[;]])
++  ],[
++    AC_DEFINE([_WIN32_WINNT], 0x0502, [Define to '0x0502' for Windows Server 2003 APIs.])
++    AC_DEFINE([NTDDI_VERSION], 0x05020000, [Define to '0x05020000' for Windows Server 2003 APIs.])
++  ])
++  AC_PREPROC_IFELSE([AC_LANG_PROGRAM(
++    [[#ifdef WINVER
++     # error WINVER already defined
++     #endif
++    ]],[[;]])
++  ],[
++    AC_DEFINE([WINVER], 0x0502, [Define to '0x0502' for Windows Server 2003 APIs.])
++  ])
+   ;;
+ esac
+ 
+-- 
+2.26.0.windows.1
+
diff --git a/contrib/src/libarchive/rules.mak b/contrib/src/libarchive/rules.mak
index 1920820bee..4592a0ec11 100644
--- a/contrib/src/libarchive/rules.mak
+++ b/contrib/src/libarchive/rules.mak
@@ -26,6 +26,7 @@ ifdef HAVE_WINSTORE
 	$(APPLY) $(SRC)/libarchive/winrt.patch
 endif
 	$(APPLY) $(SRC)/libarchive/fix-types.patch
+	$(APPLY) $(SRC)/libarchive/0005-don-t-force-windows-versions-if-they-are-set-in-the-.patch
 	$(call pkg_static,"build/pkgconfig/libarchive.pc.in")
 	$(MOVE)
 
-- 
2.26.0.windows.1

